### PR TITLE
feat: set fake k8s client before calling k8s.NewClient() to test

### PR DESCRIPTION
**Pull request by aFlyBird0**
## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [ ] I have added relevant tests

## Description
* add a new func to test functions which use `k8s.NewClient()` easily
* fix broken links

### CODE REVIEW

**Review by iyear — APPROVED** [review]
LGTM

File: docs/best-practices/gitops.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ For Chinese readers, watch this one instead:
 1. [repo-scaffolding](../plugins/repo-scaffolding.md)
 2. [jira-github](../plugins/jira-github-integ.md)
 3. [githubactions-golang](../plugins/githubactions-golang.md)
-4. [argocd](../plugins/argocd.md)
```
Comment (daniel-hutao) [inline docs/best-practices/gitops.md:22]: Please leave it wrong for now. I'll fix in an individual pr.

File: docs/best-practices/gitops.zh.md [hunk 2]
```diff
@@ -11,7 +11,7 @@
 1. [repo-scaffolding](../plugins/repo-scaffolding.zh.md)
 2. [jira-github](../plugins/jira-github-integ.zh.md)
 3. [githubactions-golang](../plugins/githubactions-golang.zh.md)
-4. [argocd](../plugins/argocd.zh.md)
```
Comment (daniel-hutao) [inline docs/best-practices/gitops.zh.md:14]: Please leave it wrong for now. I'll fix in an individual pr.

File: docs/plugins/argocdapp.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ This plugin creates an [ArgoCD Application](https://argo-cd.readthedocs.io/en/st
 **Notes:**
 
 - ArgoCD itself must have been already installed before the usage of this plugin.
-  To install ArgoCD, use the [argocd plugin](./argocd.md).
```
Comment (daniel-hutao) [inline docs/plugins/argocdapp.md:8]: Please leave it wrong for now. I'll fix in an individual pr.

**Review by daniel-hutao — CHANGES_REQUESTED** [review]

**Review by daniel-hutao — APPROVED** [review]

**Pull request merged**